### ISSUE TITLE
Default memory size of Lambda functions should be 128MB

### ISSUE THREAD

**Issue by SnowCait** [serverless/serverless#11529]
### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Use case description

There are two problems.

## AWS default is 128MB

Users don't think Serverless Framework default is different from AWS default.
So we can't realize it is needed to set `memorySize` to 128MB.
Unfortunately, we often skip over documents.

## Wasting resources

The above means many users (especially beginners like me) use functions with default 1,024MB.
Excessive memory wasts server resources.
It is not friendly to our purses and global environment.

### Proposed solution (optional)

Set default to 128MB.

I understand you want to avoid troubles without enough memory, but it causes previously mentioned problems.
I think it is more important for us to avoid useless cost because one of reasons to use serverless is low-cost.

Similar issue is #2678 which refers to template.

**Comment by medikoo** [comment]
@SnowCait 1024MB is set as default on purpose, as the AWS default provides very poor performance, which is not acceptable for most of the apps. In other words: In most cases, you need 1024MB, therefore Framework sets it as default.

**Comment by SnowCait** [comment]
@medikoo
Thank you for your prompt reply.

My hobby app which simply accesses DynamoDB (through AWS SDK for JavaScript) and fetch API works on 100-120MB.
It may need 200-300MB in the furture but 1,024MB is overspec.
Do you assume only business cases?

If Serverless Framework requires at least 100MB, I definitely feel uneasy about 128MB in most cases.
Can't you set default to 256MB?

Otherwize, could you clearly define `memorySize` on template?
We can aware lacking of memory, but can't aware overspec with no hints.

**Comment by medikoo** [comment]
> Otherwize, could you clearly define memorySize on template?

We indicate in a documentation that Framework implies 1024 as default: https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml/#general-function-settings

What exactly other improvements do you suggest?

**Comment by SnowCait** [comment]
I fortunately aware the indication but it is left up to chance.
It is difficult for many users to aware the indication from lots of document always as I've mentioned.

**Comment by SnowCait** [comment]
My suggestion is either:

- Reduce the default value
- Define `memorySize` on template (let users to aware the default is 1,024MB more easily)

**Comment by medikoo** [comment]
> Define memorySize on template (let users to aware the default is 1,024MB more easily)

What do you mean by that?

**Comment by SnowCait** [comment]
`AWS - Node.js - Starter` (`aws-node`) template creates `serverless.yml` as:

```
service: serverless

frameworkVersion: '3'

provider:
  name: aws
  runtime: nodejs14.x

functions:
  hello:
    handler: handler.hello
```

Adding `memorySize: 1024` to the `aws-node` template like this makes the default memory size clear.
Even if a user hasn't read document, he can aware the memory size is 1,024MB and will change the value to appropriate value.

```diff
 provider:
   name: aws
   runtime: nodejs14.x
+  memorySize: 1024

 functions:
   hello:
```

**Comment by medikoo** [comment]
@SnowCait if we go that route, we would have to write defaults for all the other properties, that'll make the template very noisy.

Our approach is: Make templates minimal (free of noise). Provide comprehensive documentation on all the properties and their defaults in our documentation.

**Comment by SnowCait** [comment]
I agree with you to make templates minimal but it brings users loss as cost.

To avoid the loss, I suggest to allow `memorySize` to write default exceptionally or abolish memory size default itself and write in template.
(Or remain template as is and reduce the default value.)
